modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/nhttp/HttpCoreNIOSender.java

```diff
@@ -304,27 +299,32 @@ private void removeUnwantedHeaders(MessageContext msgContext) {
      * Remove unwanted headers from the given header map.
      *
      * @param headers Header map
+     * @param nHttpConfiguration NHttp transporter base configurations
      */
-    private void removeUnwantedHeadersFromHeaderMap(Map headers) {
+    private void removeUnwantedHeadersFromHeaderMap(Map headers, NHttpConfiguration nHttpConfiguration) {
 
         Iterator iter = headers.keySet().iterator();
         while (iter.hasNext()) {
             String headerName = (String) iter.next();
             if (HTTP.CONN_DIRECTIVE.equalsIgnoreCase(headerName) ||
                 HTTP.TRANSFER_ENCODING.equalsIgnoreCase(headerName) ||
-                HTTP.DATE_HEADER.equalsIgnoreCase(headerName) ||
                 HTTP.CONTENT_TYPE.equalsIgnoreCase(headerName) ||
                 HTTP.CONTENT_LEN.equalsIgnoreCase(headerName)) {
                 iter.remove();
             }
 
-            if (!preserveServerHeader && HTTP.SERVER_HEADER.equalsIgnoreCase(headerName)) {
+            if (!nHttpConfiguration.isPreserveHttpHeader(HTTP.SERVER_HEADER) && HTTP.SERVER_HEADER.equalsIgnoreCase(headerName)) {
```

Comment:
When we move the initialization part of the isPreserveHttpHeader() to the constructor, we can have this logic as it is.